File: ui/src/main/java/example/ui/cli/DramaService.java
package example.ui.cli;

import example.common.entity.Drama;
import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import java.util.List;

@FeignClient(name = "API-GATEWAY", path = "/drama-service", fallback = DramaServiceCircuitBreaker.class)
public interface DramaService {

    @RequestMapping(path = "/dramas", method = RequestMethod.GET)
    public List<Drama> findAllDramas();

}

File: drama-service/src/main/java/example/drama/service/service/DramaService.kt
package example.drama.service.service

import example.common.entity.Drama
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional

interface DramaService {

    @Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
    fun findAllDramas(): List<Drama>

}
